### DISCUSSION TITLE
Unstable screenshots

### DISCUSSION THREAD

**Discussion by FutureKode** [percy/cli#1079] [Support]
Hello 👋🏻 

Any tips for helping Percy wait until all the UI is finished loading? We are seeing unstable screenshots where different bits of UI are not finished loading. I thought Percy was supposed to wait until network becomes idle?

Any settings I can play with to make Percy wait for all UI to load without having to add checks in the cypress tests themselves?

Any help is greatly appreciated

Thanks

**Comment by Robdel12** [comment]
Hey @FutureKode! Without a bit more info on what exactly is unstable or a link to a build we won't be able to help too much. It might be worth giving this doc a read through: https://docs.percy.io/docs/debugging-sdks#how-percy-works

> I thought Percy was supposed to wait until network becomes idle?

Percy doesn't do anything in your test suite related to that. When `percySnapshot` is called we're going to serialize the DOM at that moment. So, if the things on the page that are unstable related to the DOM state, waiting before `percySnapshot` could help. `networkIdleTimeout` is for asset discovery, which is a timeout in ms for zero network requests to be made. For example, if the timeout is set to `200`, asset discovery will wait 200ms for zero network requests to be made and then close asset discovery for that given snapshot. If you're missing assets, it'd be worth using the `--debug` flag and reading through the asset discovery logs to see what is happening to those assets.
